ledger/src/leader_schedule.rs

```diff
@@ -97,6 +123,22 @@ impl Index<u64> for LeaderSchedule {
     }
 }
 
+fn sort_stakes(stakes: &mut Vec<(&Pubkey, u64)>) {
+    // Sort first by stake. If stakes are the same, sort by pubkey to ensure a
+    // deterministic result.
+    // Note: Use unstable sort, because we dedup right after to remove the equal elements.
+    stakes.sort_unstable_by(|(l_pubkey, l_stake), (r_pubkey, r_stake)| {
+        if r_stake == l_stake {
```

Comment:
nit: you could compare the tuples directly:
```
(r_stake, r_pubkey).cmp(&(l_stake, l_pubkey))
```

Reply:
This PR just moves this `sort_stakes` function from another file, I'm fine with leaving it as is